android/app/src/main/java/org/fossasia/openevent/dbutils/RealmDataRepository.java

```diff
@@ -265,6 +266,24 @@ private void saveSessionsInRealm(final List<Session> sessions) {
                     }
                 }
 
+                SessionType sessionType = session.getSessionType();
+                if(sessionType != null) {
+                    //To prevent overwriting of previously saved values
+                    SessionType stored = transaction.where(SessionType.class).equalTo("id", sessionType.getName()).findFirst();
+
+                    if(stored != null) {
+                        session.setSessionType(stored);
+                    } else {
+                        //setting intermediate information
+                        if(sessionType.getName() == null) {
+                            sessionType.setName("");
+                        } else {
+                            sessionType.setName(sessionType.getName());
+                        }
```

Comment:
Is this block needed? We should make null safe views instead, where we check that the name is null or not. And can an event have null session type name?